From 37f91d27ff3d15f0431aba8efe1eee9dd7364d8a Mon Sep 17 00:00:00 2001
From: Arun Siluvery <arun.siluvery@linux.intel.com>
Date: Thu, 21 Jan 2016 21:43:54 +0000
Subject: [PATCH 088/143] drm/i915/gen9: Add WaOCLCoherentLineFlush

This is mainly required for future enabling of pre-emptive
command execution.

v2: explain purpose of change (Chris)

Change-Id: I0c1ac2f9ea29d69c78eeaf7841f3e339aa9e6290
Reviewed-by: Nick Hoath <nicholas.hoath@intel.com>
Cc: Dave Gordon <david.s.gordon@intel.com>
Signed-off-by: Arun Siluvery <arun.siluvery@linux.intel.com>
Link: http://patchwork.freedesktop.org/patch/msgid/1453412634-29238-9-git-send-email-arun.siluvery@linux.intel.com
Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>
---
 drivers/gpu/drm/i915/intel_ringbuffer.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

Index: current/drivers/gpu/drm/i915/intel_ringbuffer.c
===================================================================
--- current.orig/drivers/gpu/drm/i915/intel_ringbuffer.c
+++ current/drivers/gpu/drm/i915/intel_ringbuffer.c
@@ -984,6 +984,10 @@ static int gen9_init_workarounds(struct
 	/* WaDisableSTUnitPowerOptimization:skl,bxt */
 	WA_SET_BIT_MASKED(HALF_SLICE_CHICKEN2, GEN8_ST_PO_DISABLE);
 
+	/* WaOCLCoherentLineFlush:skl,bxt */
+	I915_WRITE(GEN8_L3SQCREG4, (I915_READ(GEN8_L3SQCREG4) |
+				    GEN8_LQSC_FLUSH_COHERENT_LINES));
+
 	return 0;
 }
 
